### ISSUE TITLE
Metadata

### ISSUE THREAD

**Issue by jreinhardt** [boltsparts/boltsparts#18]
Think about a way to treat metadata. Distinguish it from dimensions? How to handle the many units they will have? Other (...)?

**Comment by berndhahnebach** [comment]
like 
mm2
mm3
mm4
for all the  structural engineering data

**Comment by berndhahnebach** [comment]
from https://github.com/boltsparts/BOLTS/issues/257:

> 
> BTW a recent requirement by my company is to be able to lookup some mass per volume measurements for Australian profiles to be used in LCA calculations. So I wonder what the pros and cons are of adding extra columns:
> 
> Pros of adding columns:
> 
>     * Easy
> 
>     * Data all in one place
> 
> 
> Cons of adding columns:
> 
>     * Unclear to user which are required for dimensional (but user can read descriptions / units)
> 
>     * Constrains metadata to 1:1, cannot have 1:N relationships of metadata
> 
> 
> What's the alternative to adding columns? Groups of metadata as a separate yaml? Sharing the same key as the first column?

**Comment by berndhahnebach** [comment]
We need to tackle this down ... Give me some time to play a bit.

**Comment by Jesusbill** [comment]
I would lean towards adding more columns directly on the same yaml.
Later, if the scope of some data is specific and different, we could consider a different scope-based yaml file.
Another question is what metadata we need for each column? type | units | measure(?)

**Comment by Moult** [comment]
> I would lean towards adding more columns directly on the same yaml.

Agreed, I think most metadata is 1:1 and this is a simple solution that instantly works with boltspy. Later on if it needs to be engineered further it can be separated.

I think just type and units is necessary. "Measure" is both IFC specific and IFC agnostic, it is up to the backend to correctly interpret it.

**Comment by Jesusbill** [comment]
> I think just type and units is necessary.

Then we are (almost) _good to go_

**Comment by berndhahnebach** [comment]
I would say go for a small example and make a PR. I will have a look and give some feedback. If it is ok we merge it and could continue this way ...

**Comment by Moult** [comment]
@Jesusbill could you take a look at the stats here for open profiles https://mega.nz/file/Do5jnBiI#OACLslnku3lQTlx667lrft_3JVtFVvy9-FOfxsapxjA and let me know which stats to extract, what column names and what column ordering is preferred? If we all share the same names and order it'll make it easier for maintenance and backends.

Here are aus docs for closed profiles:

https://mega.nz/file/2tQzlSSb#z7aD3OGvKVu6xpdzR6qHvtJ40TSHzYTrBl3Fj5FavVc

And

https://mega.nz/file/etJ3iaDQ#mvvoJUGVzcwPrTMSJIFxQdbjPBg0Fhs0zDiLXqj5V0Y

**Comment by Jesusbill** [comment]
@Moult I would go with these extra stats for all profiles (as a minimum viable set for structural analysis and calculations)
![image](https://user-images.githubusercontent.com/6861582/201034060-03578265-ef93-40eb-bdcc-d0d7d0f6f5c2.png)


I am using Y and Z as the profile local directions, as customary often for structural and aligned with IFC structural and [Pset_ProfileMechanical](https://standards.buildingsmart.org/IFC/RELEASE/IFC4_1/FINAL/HTML/schema/ifcprofileresource/pset/pset_profilemechanical.htm)

Regarding relation with Pset_ProfileMechanical that would be

A &#8594; CrossSectionArea
Iy &#8594; MomentOfInertiaY
Zy &#8594; MaximumSectionModulusY / MinimumSectionModulusY
Sy &#8594; PlasticShapeFactorY * MinimumSectionModulusY ; in other words PlasticShapeFactorY = Sy / Zy
Iz ...
Zz ...
Sz ...
It &#8594; TorsionalConstantX
Cw &#8594; WarpingConstant

For non symmetric profiles I would add the geometric center as well
The order above and in the paper looks ok for the properties, although I would not rely on it strongly. Ideally adding extra properties across the two directions should group together with the rest, and not appended at the end.

edit: updated figure

**Comment by Moult** [comment]
Thanks @Jesusbill ! If you add an example for a profile, we can follow your lead.

**Comment by Jesusbill** [comment]
@berndhahnebach @Moult I added the additional properties for two groups with SI and imperial units.
I have used the symbol `^` for the power in units, other than that for now symbol names and order is as shown [above](https://github.com/boltsparts/boltsparts/issues/18).

**Comment by Jesusbill** [comment]
hi @berndhahnebach what do you reckon about the additions I made, anything not convincing you?
I am not entirely convinced by some of the type names but I wanted to start from somewhere.

**Comment by berndhahnebach** [comment]
sorry guys, I've had very busy Christmas and following days ... No updates on BOLTS for to long ... I will be on ski vacation the weed after this week, no vist in Bruessels to Fosdem :-( 
After this I will come back to BOLTS too.

cheers bernd
